jazkarta/shop/cart.py

```diff
@@ -185,7 +185,6 @@ def from_request(cls, request, user_id=None, browser_id=None):
                 else:
                     # create a new cart
                     data = PersistentMapping()
-                storage.set_shop_data([user_id, 'cart'], data)
```

Comment:
Is this directly related to the functionality change, or was this redundant in the original code, since the `Cart` constructor is later called with `data` as an argument?

Reply:
My feeling here was that there's no need to persist the (often empty) cart data whenever it's retrieved. Persisting it when items are added/removed/quantities changed/cleared should be sufficient.

I believe this was redundant in the original code, and it also ensured that every browser session that resulted in a cart lookup (possibly every session ever), persisted a (probably empty) cart in the data structure. Without this change, removing the cart once emptied was pointless because the cart would always be recreated when retrieved. I suspect it's also bad for ConfilctErrors and auto-CSRF protection issues for the same reason.